go/store/prolly/tuple_range.go

```diff
@@ -167,18 +173,161 @@ func (r Range) Matches(t val.Tuple) bool {
 	return true
 }
 
-func (r Range) IsPointLookup(desc val.TupleDesc) bool {
+func (r Range) IsStrictKeyLookup(desc val.TupleDesc) bool {
+	// a strict key is a set of non-nil equality restrictions covering every field of a unique index
 	if len(r.Fields) < len(desc.Types) {
 		return false
 	}
 	for i := range r.Fields {
-		if !r.Fields[i].Exact {
+		if !r.Fields[i].BoundsAreEqual {
 			return false
 		}
 	}
 	return true
 }
 
+// KeyRangeLookup will return a stop key and true if the range can be scanned
+// from a start to stop tuple. Otherwise, return a nil key and false. A range
+// can be key range scanned if the prefix is exact, and the final field is
+// numeric or string. The stop key adds +1 to a numeric final field, and appends
+// '0' to a string final field.
+// TODO: support non-exact final field, and use range upper bound?
+func (r Range) KeyRangeLookup(pool pool.BuffPool) (val.Tuple, bool) {
+	if r.Tup == nil {
+		return nil, false
+	}
+	n := len(r.Fields) - 1
+	for i := range r.Fields {
+		if r.Fields[i].Lo.Value == nil {
+			if r.Fields[i].Hi.Value != nil {
+				return nil, false
+			}
+			n = i - 1
+			break
+		}
+		if !r.Fields[i].BoundsAreEqual {
+			return nil, false
+		}
+	}
+
+	if n < 0 {
+		// ex: range scan
+		return nil, false
+	}
+
+	for _, typ := range r.Desc.Types[n+1:] {
+		if !typ.Nullable {
+			// this is checked separately because fulltext descriptors
+			// do not match field lengths sometimes
+			// todo: why?
```

Comment:
Can you create an issue for this and link to it?

Reply:
yep https://github.com/dolthub/dolt/issues/8029